crates/bevy_ecs/src/query/mod.rs

```diff
@@ -144,18 +144,30 @@ mod tests {
         assert_eq!(values.iter(&world).size_hint().1.unwrap(), n);
         assert_eq!(values.iter(&world).len(), n);
         assert_eq!(values.iter(&world).count(), n);
+        let mut iterator = values.iter(&world);
```

Comment:
The tests are not as exhaustive as I'd like, but I'm waiting for https://github.com/bevyengine/bevy/pull/5148 to improve them without too much duplication.